src/site/topic22.rst

```diff
@@ -263,6 +263,106 @@ Binary Tree Node
 Linked Binary Tree
 ^^^^^^^^^^^^^^^^^^
 
+* Although there will be no implementation of a general ``BinaryTree``, we can discuss what some specific tree based algorithms
+
+**Size**
+
+* If we have some arbitrary binary tree and do not know it's size (and it has no size field), how would we count the number of elements?
+
+.. code-block:: java
+    :linenos:
+
+    public int size() {
+        return size(root);
+    }
+
+    private int size(Node<T> current) {
+        if (current == null) {
+            return 0;
+        } else {
+            return 1 + size((current.getLeft())) + size(current.getRight());
+        }
+    }
+
+* The idea here is, if the current node exists, the the size of the (sub)tree will be 1 + the size of the left subtree + the size of the right subtree
+* Here we also make use of a public helper method that always starts the recursive method at the root
+
+* What is the computational complexity of ``size()``?
+    * :math:`O(n)`, where :math:`n` is the number of nodes in the tree
+
+**Contains**
+
+* If we have an arbitrary binary tree and we want to search it for a specific element
+
+.. code-block:: java
+    :linenos:
+
+    public boolean contains(T element) {
```

Comment:
`element` -> `needle` ([as mentioned in previous topics](https://github.com/jameshughes89/cs102/blame/main/src/site/topic20.rst#L18))

Reply:
changed in the code, but will keep element in cases like: 

> If we have an arbitrary binary tree and we want to search it for a specific element

Reply:
I don't see why. An array is as arbitrary a collection as a tree. Search is search.

Reply:
Ah, I misunderstood. You meant in the text, you'd use the word element. That's fine, makes sense.

Reply:
Brevity is perhaps not the best way to communicate information.

Reply:
🤔

Reply:
:thinking: :thinking: :thinking: